### PR TITLE
TD-3161 Model Button Edit Styles

### PR DESCRIPTION
**Pull request by jzhelyazkovtvm**
Closes [TD-3161](https://sce.myjetbrains.com/youtrack/issue/TD-3161/Edit-the-styles-of-ModelButton-in-React-UI)

## Changes

Updated ModelButton to show status icon on top right of the button. Borders and background color also updated to match the status colors defined in figma, with new hover behaviour as well. Popup button also has now outlined style, matching the colors and background from the ModelButton component.
The style of the component when no status is passed remains the same.

A brief description of what this pull request solves / introduces.

- Updated ModelButton with status icon and background + borders
- Updated ModelButtonPopup with status icon with background + borders
- Old styles for when button is disabled or no children are still the same

Here is the [ticket](https://github.com/IPG-Automotive-UK/virto/pull/1133), where these changes are visible in VIRTO FLEET and MODEL with pre-release.

## Dependencies

N/A

## UI/UX
Before
image

After
image
image
image
image

[Figma prototype](https://www.figma.com/design/aFUbF9gTc2OfeLQnSf6BAF/VIRTO.BUILD?node-id=7574-26411&node-type=frame&t=DNhpHoaLpbedAS3p-0)

## Testing notes

The Model Button component should have the icon for the specific status in top right corner. The colors must match the colors from figma prototype above. The popup button should also match Model button border color and the checkmark inside.
Onhover effects, including background and and borders of the icons must match the onhover state in figma.
The border in ModelButtonPopup should be the same(2px) as in ModelButton.
When no status is passed to the component, there should not be icon in top right corner and changes of the colors(border or background).
The image inside of ModelButton should not be changed on hover and should stay black or white depending on the mode of the theme. When no status is passed then, we should be able to see the primary color changing  the white/black on hover.

## Author checklist

Before I request a review:

- [x] I have reviewed my own code-diff.
- [x] I have tested the changes in Docker / a deploy-preview.
- [x] I have assigned the PR to myself or an appropriate delegate.
- [x] I have added the relevant labels to the PR.
- [x] I have included appropriate tests.
- [x] I have checked that the Lint and Test workflows pass on Github.
- ~[ ] I have populated the deploy-preview with relevant test data.~
- ~[ ] I have tagged a UI/UX designer for review if this PR includes UI/UX changes.~

### CODE REVIEW

**Review by m4manjesh — CHANGES_REQUESTED** [review]
@jzhelyazkovtvm  this PR inlcudes non related changed to this PR, please fix it. Also please update the testing notes.

**Comment by jzhelyazkovtvm** [comment]
> @jzhelyazkovtvm this PR inlcudes non related changes, please fix it. Please update the testing notes with what needed to be tested, Author checklists are not checked correclty, this PR still created a deply-preview, if you tested it there the second point should be checked. I see that @Sowbhagya-ipg approved the UI in another PR, but i suggest in future tag her here as well so can fix any UI issue upfront

We agreed on Tagging UX designer for VIRTO Model and Fleet, where the buttons are rendered with pre-release of RUI.

**Comment by m4manjesh** [comment]
> > @jzhelyazkovtvm this PR inlcudes non related changes, please fix it. Please update the testing notes with what needed to be tested, Author checklists are not checked correclty, this PR still created a deply-preview, if you tested it there the second point should be checked. I see that @Sowbhagya-ipg approved the UI in another PR, but i suggest in future tag her here as well so can fix any UI issue upfront
> 
> We agreed on Tagging UX designer for VIRTO Model and Fleet, where the buttons are rendered with pre-release of RUI.

Make sure you share the testing PR link here, ok in this case as i'm reviewing the other one.

File: src/ModelButton/ModelButton.tsx [hunk 1]
```diff
@@ -305,3 +355,25 @@ const ModelButtonPopup = ({
     </React.Fragment>
   );
 };
+
+/** Get the correct icon button background color, according to the theme and hover behaviour  */
+const getCurrentIconBackgroundColor = (
+  isHover: boolean,
+  status: string,
```
Comment (m4manjesh) [inline src/ModelButton/ModelButton.tsx:362]: props types should live in `ModelButton.types.ts`

**Review by m4manjesh — CHANGES_REQUESTED** [review]
not directly realted but i see we are using `PaperProps` for the popover which is depricated, please use the following instead `slotProps={{ paper: { sx: { borderRadius: 2, padding: 2 } } }}`

Component looks good, some small suggestions

File: src/ModelButton/ModelButton.test.tsx [hunk 2]
```diff
@@ -49,6 +50,50 @@ describe("ModelButton", () => {
     expect(svgElement).toHaveAttribute("stroke-width", "2");
   });
 
+  // test when status different than 'none' icon is rendered
+  test("does show status icon", () => {
+    render(<ModelButton status="error" />);
```
Comment (m4manjesh) [inline src/ModelButton/ModelButton.test.tsx:55]: Test below already doing the same. You can test no icon renderes if the status is `non`

File: src/ModelButton/ModelButton.test.tsx [hunk 3]
```diff
@@ -49,6 +50,50 @@ describe("ModelButton", () => {
...
+    "renrers correct background based on the status",
+    (status, color) => {
+      render(<ModelButton status={status} />);
+      const svgElement = screen.getByTestId("background");
+      expect(svgElement).toHaveAttribute("fill", alpha(color, 0.04));
+    }
+  );
+
+  // test when status different than 'none' appropriate background is added to icon button
+  test("does apply background color", () => {
```
Comment (m4manjesh) [inline src/ModelButton/ModelButton.test.tsx:90]: Test above is going the same, instead you can test case `none`

**Comment by jzhelyazkovtvm** [comment]
> > > @jzhelyazkovtvm this PR inlcudes non related changes, please fix it. Please update the testing notes with what needed to be tested, Author checklists are not checked correclty, this PR still created a deply-preview, if you tested it there the second point should be checked. I see that @Sowbhagya-ipg approved the UI in another PR, but i suggest in future tag her here as well so can fix any UI issue upfront
> > 
> > 
> > We agreed on Tagging UX designer for VIRTO Model and Fleet, where the buttons are rendered with pre-release of RUI.
> 
> Make sure you share the testing PR link here, ok in this case as i'm reviewing the other one.

image

**Comment by jzhelyazkovtvm** [comment]
@m4manjesh  we will not need pre-release this time, since there are no changes in the UI

**Review by m4manjesh — APPROVED** [review]
LGTM

**Pull request merged**